zkevm-circuits/src/blake2b/src/main.rs

```diff
@@ -0,0 +1,442 @@
+extern crate halo2_proofs;
+use halo2_proofs::{arithmetic::FieldExt, circuit::*, plonk::*, poly::Rotation, dev::MockProver, halo2curves::bn256::Fr};
+use std::{marker::PhantomData, array, convert::TryInto};
+
+fn gf<F:FieldExt>(n: u64) -> Expression<F> {
+    Expression::Constant(F::from(n))
+}
+
+fn array_refs_from_slice<T, const S: usize, const L: usize>(slice: &[T]) -> [&[T; S]; L] {
+    array::from_fn(|i| slice[S * i..][..S].try_into().unwrap())
+}
+
+fn byte_cells_to_number<F:FieldExt>(meta: &mut VirtualCells<F>, cells: &[ACell<F>]) -> Expression<F> {
+    cells.iter().map(|c| c.express(meta)).rfold(gf(0), |e, b| { e * gf(256) + b })
+}
+
+fn set_byte_witnesses<F:FieldExt, const N:usize>(region: &mut Region<F>, name: &str, 
+    row: usize, cells: &[ACell<F>; N], bytes: &[u8; N]) -> Result<(), Error> {
+        for (c, b) in cells.iter().zip(bytes.iter())  { 
+            region.assign_advice(|| name, c.column, row + (c.offset as usize), || Value::known(F::from(*b as u64)))?;
+        }
+        return Ok(());
```

Comment:
```suggestion
        Ok(())
```
I rarely see `return` used in rust, so I think a good habit to not use it where possible